src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -179,9 +182,21 @@ export const portfolioApi = createApi({
         try {
           const adapter = chainAdapters.get(chainId)
           if (!adapter) throw new Error(`no adapter for ${chainId} not available`)
-          const portfolioAccounts = { [pubkey]: await adapter.getAccount(pubkey) }
+          // "Fetch" the query leveraging the existing cached data
+          const portfolioAccounts = {
+            [pubkey]: await queryClient.fetchQuery({
+              ...accountManagement.getAccount(accountId),
+              staleTime: GET_ACCOUNT_STALE_TIME,
+              // Never garbage collect me, I'm a special snowflake
+              gcTime: Infinity,
+            }),
+          }
+
           const nftCollectionsById = selectNftCollections(state)
 
+          // Prefetch smart contract checks - do *not* await/.then() me, this is only for the purpose of having this cached later
```

Comment:
Precisely what the comment says - we run this query as early as possible in `getAccount()` RTK query so that it's cached later on.